internal/worker/server.go

```diff
@@ -290,6 +258,98 @@ func (s *Server) JobDependencyChainErrors(id uuid.UUID) (*clienterrors.Error, er
 	return nil, nil
 }
 
+// Optimized way to get the error out of the OSBuildJobResult without having to fetch the entire result blob.
+// Each fields are fetched independently in subsequent requests to the database.
+// The logic is the same as for OSBuildJobInfo regarding error handling.
+func (s *Server) GetOSBuildJobError(id uuid.UUID, finished time.Time) (jobError *clienterrors.Error, err error) {
+	err = s.jobs.QueryJsonField(id, "result", "job_error", &jobError)
+	if err != nil {
+		return nil, err
+	}
+	if jobError == nil && !finished.IsZero() {
+		has_output, err := s.jobs.TestJsonFieldExists(id, "result", "osbuild_output")
+		if err != nil {
+			return nil, err
+		}
+		if !has_output {
+			return clienterrors.WorkerClientError(clienterrors.ErrorBuildJob, "osbuild build failed", nil), nil
+		}
+		var osbuild_output_error string
+		err = s.jobs.QueryJsonField(id, "result", "osbuild_output.error", &osbuild_output_error)
+		if err != nil {
+			return nil, err
+		}
+		if osbuild_output_error != "" {
+			return clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, osbuild_output_error, nil), nil
+		}
+		var mock_result OSBuildJobResult
+		err = s.jobs.QueryJsonField(id, "result", "target_results", &mock_result.TargetResults)
+		if err != nil {
+			return nil, err
+		}
+		if len(mock_result.TargetErrors()) > 0 {
+			return clienterrors.WorkerClientError(clienterrors.ErrorTargetError, "at least one target failed", mock_result.TargetErrors()), nil
+		}
```

Comment:
wdym ? I feel like I'm missing something because I've copied paste this behavior from the previous function

Reply:
Ah right, TargetResult can contains a TargetError, nvm yea, resolving.

Maybe instead of mock_result we can have a better name like targetResults though.